scripts/signup.js

```diff
@@ -0,0 +1,82 @@
+//@ts-check
+import {
+  EMAIL_REGEX,
+  INPUT_IDS,
+  PASSWORD_MESSAGE,
+  PASSWORD_REGEX,
+  USERS,
+  signUpAction
+} from './constant/signConfig.js';
+import { DOMHandler, InputHandler } from './utils/element.js';
+import { SignHandler } from './utils/sign.js';
+
+const {
+  emailElementId,
+  passwordElementId,
+  passwordCheckElementId,
+  emailErrorElementId,
+  passwordErrorElementId,
+  passwordCheckErrorElementId,
+  loginFormId,
+  eyeImagePasswordId,
+  eyeImagePasswordCheckId
+} = INPUT_IDS;
+
+/** @type {HTMLInputElement} emailInput*/
+const emailElement = DOMHandler.getById(emailElementId);
+/** @type {HTMLInputElement} passwordElement*/
+const passwordElement = DOMHandler.getById(passwordElementId);
+/** @type {HTMLInputElement} loginButton*/
+const passwordCheckElement = DOMHandler.getById(passwordCheckElementId);
+/** @type {HTMLFormElement} loginForm*/
+const loginForm = DOMHandler.getById(loginFormId);
+/** @type {HTMLImageElement} eyeImagePassword*/
+const eyeImagePassword = DOMHandler.getById(eyeImagePasswordId);
+/** @type {HTMLImageElement} eyeImagePasswordCheck*/
+const eyeImagePasswordCheck = DOMHandler.getById(eyeImagePasswordCheckId);
+DOMHandler.addPAfterElement(emailElement, emailErrorElementId, 'error-text');
+DOMHandler.addPAfterElement(passwordElement, passwordErrorElementId, 'error-text');
+DOMHandler.addPAfterElement(passwordCheckElement, passwordCheckErrorElementId, 'error-text');
+/** @type {HTMLElement} emailErrorElement*/
+const emailErrorElement = DOMHandler.getById(emailErrorElementId);
+/** @type {HTMLElement} passwordErrorElement*/
+const passwordErrorElement = DOMHandler.getById(passwordErrorElementId);
+/** @type {HTMLElement} passwordCheckErrorElement*/
+const passwordCheckErrorElement = DOMHandler.getById(passwordCheckErrorElementId);
+
+const handlePasswordMatch = () => {
+  if (InputHandler.isMatchElement(passwordElement, passwordCheckElement)) {
+    passwordCheckErrorElement.classList.add('hidden');
+    return;
+  }
+  SignHandler.showErrorMessage(passwordCheckElement, passwordCheckErrorElement, PASSWORD_MESSAGE.match);
+};
+/** @param {Event} event */
+const handleSignUp = event => {
+  event.preventDefault();
+  emailElement.blur();
+  passwordElement.blur();
+  passwordCheckElement.blur();
+
+  const checkEmail =
+    InputHandler.isMatchRegEx(emailElement, EMAIL_REGEX) && !SignHandler.isExistEmail(USERS, emailElement);
+  const checkPassword =
+    InputHandler.isMatchRegEx(passwordElement, PASSWORD_REGEX) &&
+    InputHandler.isMatchElement(passwordElement, passwordCheckElement);
+  if (checkEmail && checkPassword) return signUpAction();
+};
```

Comment:
```suggestion
  if (checkEmail && checkPassword) {
      signUpAction();
      return;
  }
};
```

함수 호출을 리턴하는 것보다 함수를 호출 한 후에 함수를 종료하는게 더 흐름상 맞아보이네요!

Reply:
짠!
마지막 이라 제거했는데 제거해도 괜찮을까용??
https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission/pull/363/commits/a3f0e7b9eeb86ff13b7ab198308c2da47479bba1